hnn_core/gui/gui.py

```diff
@@ -628,16 +803,73 @@ def _clear_params(b):
+            with self._log_out:
+                assert new_key in self.analysis_config['gui_config'][
+                    'fig_templates'].keys()
+                assert self.fig_titles.selected_index == len(
+                    self.data['figure_trees'])
+            _init_plot_configs(self._plot_configs,
+                               self.simulation_data,
+                               self.vis_plot_tree,
+                               self.analysis_config,
+                               use_template=True,
+                               render=True)
+
+        self.default_viz_config_selection.observe(
+            _handle_viz_config_layout_change,
+            "value"
+        )
+
+        def _handle_fig_change(b):
+            fig_idx = b['new']
+            n_figs = len(self.figure_trees)
+            if fig_idx == n_figs:
+                self.default_viz_config_selection.disabled = False
+                self.run_plotting_button.description = add_fig_description
+                # selected current
+                self.run_plotting_button.description = add_fig_description
+                _init_plot_configs(self._plot_configs, self.simulation_data,
+                                   self.vis_plot_tree, self.analysis_config,
+                                   render=True)
+            else:
+                self.default_viz_config_selection.disabled = True
+                fig_idx = b['new']
+                if fig_idx is not None:
+                    assert 0 <= fig_idx < len(self.figure_trees)
+                    self.run_plotting_button.description = f"Save \
+                        Fig.{fig_idx+1}"
+                    _init_plot_configs(self._plot_configs,
+                                       self.simulation_data,
+                                       self.figure_trees[fig_idx],
+                                       self.analysis_config, render=True)
+
+        self.fig_titles.observe(_handle_fig_change, names='selected_index')
+
+        def _on_plot_clicked(_):
+            if self.fig_titles.selected_index == len(self.figure_trees):
+                # add a figure
+                try:
+                    self.figure_trees.append(copy.deepcopy(self.vis_plot_tree))
+                    n_figs = len(self.figure_trees)
+                    self.fig_titles.selected_index = None
+                    self.fig_titles.children = [
+                        self._visualization_window for _ in range(n_figs + 1)
+                    ]
+                    for idx in range(n_figs + 1):
+                        self.fig_titles.set_title(
+                            idx, f'Fig.{idx+1}' if idx < n_figs else 'current')
+
+                    # select the latest created fig
+                    self.fig_titles.selected_index = n_figs - 1
+
+                    plot_sim_data(self.simulation_data, self.vis_plot_tree,
+                                  self.analysis_config)
+                except Exception as e:
```

Comment:
We should add a proper exception handler like here: https://ipywidgets.readthedocs.io/en/stable/examples/Output%20Widget.html#Integrating-output-widgets-with-the-logging-module